include/envoy/router/route_config_update_receiver.h

```diff
@@ -82,6 +82,11 @@ class RouteConfigUpdateReceiver {
    */
   virtual const envoy::config::route::v3::RouteConfiguration& routeConfiguration() PURE;
 
+  /**
+   * @return Router::ConfigConstSharedPtr a route configuration after the latest update.
+   */
+  virtual ConfigConstSharedPtr config() const PURE;
```

Comment:
What are the semeantics of this vs. `routeConfiguration()`? They seems to convey similar looking things.

Reply:
IIUC this is only used inside `RdsRouteConfigProviderImpl::onConfigUpdate()`. If you are not planning to use it in other places, it can remain a local variable there.

Reply:
> What are the semeantics of this vs. routeConfiguration()

`config()` returns a parsed (and validated) `Router::ConfigImpl`, the latter returns RouteConfiguration protobuf. I don't like the name `config` for the method, but couldn't think of anything short but descriptive.

Reply:
> IIUC this is only used inside RdsRouteConfigProviderImpl::onConfigUpdate(). If you are not planning to use it in other places, it can remain a local variable there.

It's possible, but that would require parsing the RouteConfiguration protobuf again, which isn't cheap.

Reply:
I realised I should probably give a gist of why I went with this implementation for the fix.

During `Router::ConfigImpl` instantiation, while RouteConfiguration protobuf is being parsed, a few exceptions can be raised (please see my comment here: https://github.com/envoyproxy/envoy/issues/14918#issuecomment-781578482).

Instead of trying to add additional validations in GrpxMux, which would then need to be kept in sync with Router::ConfigImpl, I decided to instantiate Router::ConfigImpl sooner (here: https://github.com/envoyproxy/envoy/pull/15110/files#diff-970dbedbda5c661c37ede4adabe63ea9f69fb54455d67339f23db5258ae62483R30), and keep the result around.

Should ctor raise an exception, VHDS state won't be updated, and subsequent updates will apply cleanly.

Reply:
Yeah, I agree with the general approach. We need comments explaining how `config()` and `routeConfiguration()` relate, and probably better names. Suggest `internalConfig()` and `protoConfig()` or something like that, i.e. convey they are the exact same config with different views?

Reply:
Thanks for the explanation. Makes sense to me.

Reply:
> We need comments explaining how config() and routeConfiguration() relate, and probably better names. Suggest internalConfig() and protoConfig()

Renamed interface methods to `parsedConfiguration()` and `protobufConfiguration()` respectively. Updated the docs too.